Added releasemanagement module to angular and added camunda process engine

jandestiny opened the pull request:
- Created new folder "/camunda" for camunda process engine, which is in out-of-the-box state
- Added multiple components to angular project under /app/components/releasemanagement
- Added according angular services under /app/services/releasemanagement

This is still a work in progress. Commit history was unfortunately lost.

jandestiny committed:
creating related history between branch and main

jandestiny committed:
Nav: Fixed user default image, RMS: Implemented dialog for editing tasks -> TODO: add functionality

jandestiny committed:
ng: made tasks submittable and variables editable, expanded camunda api connector methods, added instance options; camunda: changed bpmn process

RammboNr5 commented on angular/src/app/components/release-management/release-detail/release-detail.component.ts:
```diff
@@ -0,0 +1,129 @@
+        this.isLoading = false;
+        },
+        error => {
+          console.warn("Keine Variablen gefunden, Instanz womöglich gelöscht", error)
```
isLoading stays true

jandestiny replied:
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny committed:
fixed several loading animations, refactored httpResponseHandling

jandestiny commented on angular/src/app/components/navigation/navigation.component.ts:
```diff
@@ -9,8 +9,8 @@ import { map, shareReplay } from 'rxjs/operators';
 export class NavigationComponent {
-  title: String = "ALLEGRO-Cockpit";
-  
+  title: String = "***REMOVED***-Cockpit";
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny commented on angular/src/app/components/release-management/create-new-release/create-new-release.component.ts:
```diff
@@ -0,0 +1,98 @@
+
+  submitNewReleaseForm() {
+
+    if (this.newReleaseForm.valid) {
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny commented on angular/src/app/components/release-management/create-new-release/create-new-release.component.ts:
```diff
@@ -0,0 +1,98 @@
+
+            console.error("Lieferung konnte nicht erstellt werden.")
+
+            this.notify.open(`Lieferung koonnnte nicht erstellt werden, bitte prüfen Sie die eingegebenen Werte. ${response.statusText}`,
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny commented on angular/src/app/components/release-management/create-new-release/create-new-release.component.ts:
```diff
@@ -0,0 +1,98 @@
+
+      this.camunda.createNewReleaseWithVariables(this.startVariables)
+        .subscribe(response => {
+          if (response.ok) {
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny commented on angular/src/app/components/release-management/edit-task/edit-task.component.ts:
```diff
@@ -0,0 +1,73 @@
+  //Could be used later, if Camunda Embedded/External Forms are implemented.
+  // @ViewChild('taskFormHtml') taskFormHtml: ElementRef;
+
+  isLoading: boolean = true;
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny commented on angular/src/app/components/release-management/create-new-release/create-new-release.component.ts:
```diff
@@ -0,0 +1,98 @@
+            this.router.navigateByUrl("/releases")
+
+          } else {
+
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny commented on angular/src/app/components/release-management/edit-task/edit-task.component.ts:
```diff
@@ -0,0 +1,73 @@
+        this.task.variables.set(variableName, newReleaseVariable);
+        this.isLoading = false;
+      },
+        error => {
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny commented on angular/src/app/components/release-management/edit-task/edit-task.component.ts:
```diff
@@ -0,0 +1,73 @@
+          this.dialogReference.close()
+        },
+        error => {
+          //TODO: Add error notification
```
fixed in latest [commit ](https://github.com/Sultanow/js_menus/pull/60/commits/158507961befce32f22c37648fd6f434a538cbca)

jandestiny committed:
ng: added functionality to add variables to releases, changed release card titel to be more descriptive

jandestiny committed:
bpmn: fixed minor bug in process diagram

jandestiny committed:
Update WorkflowTest.java

Changed test to fit default-release-management-process and eliminate build error

jandestiny committed:
Update Application.java

removed testing endpoint

jandestiny committed:
Add files via upload

Added readme for Camunda

jandestiny committed:
Update README.md

jandestiny committed:
Update README.md

jandestiny committed:
Update README.md

jandestiny committed:
Update README.md

jandestiny committed:
Create README.md

readme file for release-management-component

jandestiny committed:
Update README.md

jandestiny committed:
Update README.md

jandestiny committed:
Update README.md

jandestiny committed:
Add files via upload

Added release-management-readme

jandestiny committed:
Update Release-Management-System-README.md

jandestiny committed:
Update README.md

jandestiny committed:
Update README.md

jandestiny committed:
Update Application.java

jandestiny committed:
Update README.md

RammboNr5 commented on angular/src/app/components/release-management/edit-task/edit-task.component.html:
```diff
@@ -0,0 +1,71 @@
+<div class="grid-container">
+  <mat-progress-bar *ngIf="isLoading" mode="indeterminate"></mat-progress-bar>
+  <h1 mat-dialog-title class="mat-h1">{{ task.name }}</h1>
+  <h3 mat-dialog-title class="mat-h3">Bitte die nachfolgenden Felder ausfüllen, um die Bearbeitung abschließen.</h3>
```
abzuschließen

jandestiny replied:
done

RammboNr5 commented on angular/src/app/components/release-management/edit-task/edit-task.component.html:
```diff
@@ -0,0 +1,71 @@
+<ng-container mat-dialog-title *ngIf="submitFormError">
+  <h3><mat-icon>warning</mat-icon>Fehler</h3>
+  <span>Bitte stellen Sie sicher, dass alle Felder ordnungsgemäß befüllt worden sind.</span><br>
+  <span>Leere Felder sind mit <mat-icon color="primary">info</mat-icon> markiert.</span>
```
gilt das für alle oder nur für Pflichtfelder? So wie es hier steht sind alle gemeint

jandestiny replied:
Nur Pflichtfelder, done

RammboNr5 commented on angular/src/app/components/release-management/edit-task/edit-task.component.ts:
```diff
@@ -0,0 +1,77 @@
+  ) { }
+
+  ngOnInit(): void {
+    // this.camunda.getTaskFormByTaskId(this.task.id)
```
remove commented code

jandestiny replied:
done

RammboNr5 commented on angular/src/app/components/release-management/release-detail/release-detail.component.ts:
```diff
@@ -0,0 +1,130 @@
+        this.isLoading = false;
+      },
+        error => {
+        console.error("Variable konnte nicht aktualisiert werden", error)
```
this.isLoading = false

jandestiny replied:
done

RammboNr5 commented on angular/src/app/components/release-management/release-management.component.ts:
```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { VersionDataMockService } from '../../services/releasemanagement/version-data-mock.service'
+// import { CamundaApiMockService } from '../../services/releasemanagement/camunda-api-mock.service'
```
delete this

RammboNr5 commented on angular/src/app/services/releasemanagement/camunda-api-connector.service.ts:
```diff
@@ -0,0 +1,175 @@
+
+    //Task methods
+
+  // private getCurrentTasks(): Observable<CamundaTask[]> {
```
delete commented code

jandestiny replied:
done

RammboNr5 commented on camunda/README.md:
```diff
@@ -0,0 +1,56 @@
+
+# Release-Management-System: Workflow Engine
+
+Dieser Ordner enthält eine Instanz von Camunda Platform 7, die im Release-Management-System des Informations-Cockpits genutzt wird.
```
IT-Cockpit

jandestiny committed:
fixed typos and removed commented code
